frb_codegen/src/generator/dart/ty_general_list.rs

```diff
@@ -12,15 +12,33 @@ impl TypeDartGeneratorTrait for TypeGeneralListGenerator<'_> {
         let context = self.context.config.block_index;
         let inner = self.ir.inner.safe_ident();
 
-        Acc {
-            io: Some(format!(
+        let io = if self.ir.inner.is_primitive() {
+            // Handle primitive enums list.
+            // This is similar to `StringList` in
+            // `frb_codegen/src/generator/dart/ty_delegate.rs`
+            format!(
+                "final ans = inner.new_{}_{}(raw.length);
+                for (var i = 0; i < raw.length; i++){{
+                    ans.ref.ptr[i] = api2wire_{}(raw[i]);
+                }}
+                return ans;",
+                self.ir.safe_ident(),
+                self.context.config.block_index,
+                self.ir.inner.safe_ident()
+            )
+        } else {
+            format!(
                 "final ans = inner.new_{ident}_{context}(raw.length);
                 for (var i = 0; i < raw.length; ++i) {{
                     _api_fill_to_wire_{inner}(raw[i], ans.ref.ptr[i]);
                 }}
                 return ans;
                 "
-            )),
+            )
+        };
```

Comment:
nit: looks like the if vs else branch only differs by the line inside for loop (ans.ref.ptr[i] vs api_fill_to_wire). Thus maybe we can make this code a bit more concise, by extracting out common parts and only write if/else for the different thing